sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs

```diff
@@ -127,6 +127,12 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
                     }
                 }
 
+                if (message.HasResponse)
+                {
+                    // Dispose the content stream to free up a connection if the request has any
+                    message.Response.ContentStream?.Dispose();
```

Comment:
Can't response be null for some custom transports?

Reply:
No, response has to be set after transport returned from Process[Async] method. This is one of the core assumptions of our pipeline.